### ISSUE TITLE
Is the Affine layer like Flatten in Keras for CNN since it is linear?

### ISSUE THREAD

**Issue by pantherso48** [NervanaSystems/neon#419]
I am looking through https://github.com/NervanaSystems/meetup/blob/master/cifar_example.ipynb as an example for writing my own CNN with my own data (cats and dogs images).

I am trying to mimic the same CNN I have created with Keras but classifying images and when I need to take the filtered feature map and flatten it in a layer I am looking to Affine layer to achieve this but not sure and could use guidance.

`from keras.preprocessing.image import ImageDataGenerator

train_datagen = ImageDataGenerator(rescale = 1./255,
                                   shear_range = 0.2,
                                   zoom_range = 0.2,
                                   horizontal_flip = True)
#

test_datagen = ImageDataGenerator(rescale = 1./255)

training_set = train_datagen.flow_from_directory('/data/datasets/catsanddogs/training_set',
                                                 target_size = (64, 64), #need dimensions expected by CNN
                                                 batch_size = 32,
                                                 class_mode = 'binary')

test_set = test_datagen.flow_from_directory('/data/datasets/catsanddogs/test_set',
                                            target_size = (64, 64), #need dimensions expected by CNN
                                            batch_size = 32,
                                            class_mode = 'binary')`

That is the code to setup my training and test sets to give you an idea of the data setup and here is the code for the CNN:
`# Now we create a model by assembling some layers
from neon.layers import Conv, Affine, Pooling
from neon.initializers import Uniform
from neon.transforms.activation import Rectlin, Softmax
init_uni = Uniform(low=-0.1, high=0.1)
layers = [Conv(fshape=(5,5,16), init=init_uni, activation=Rectlin()),
          Pooling(fshape=2, strides=2),
          Conv(fshape=(5,5,32), init=init_uni, activation=Rectlin()),
          Pooling(fshape=2, strides=2),
          Affine(nout=500, init=init_uni, activation=Rectlin()),
          Affine(nout=10, init=init_uni, activation=Softmax())]

# set up model
from neon.models import Model
model = Model(layers)`
